src/coreComponents/constitutive/diffusion/ConstantDiffusion.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ConstantDiffusion.cpp
+ */
+
+#include "ConstantDiffusion.hpp"
+
+namespace geos
+{
+
+using namespace dataRepository;
+
+namespace constitutive
+{
+
+ConstantDiffusion::ConstantDiffusion( string const & name, Group * const parent ):
+  DiffusionBase( name, parent )
+{
+  registerWrapper( viewKeyStruct::diffusivityComponentsString(), &m_diffusivityComponents ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setRestartFlags( RestartFlags::NO_WRITE ).
+    setDescription( "xx, yy, and zz components of a diffusivity tensor [m^2/s]" );
+}
+
+std::unique_ptr< ConstitutiveBase >
+ConstantDiffusion::deliverClone( string const & name,
+                                 Group * const parent ) const
+{
+  return DiffusionBase::deliverClone( name, parent );
+}
+
+void ConstantDiffusion::allocateConstitutiveData( dataRepository::Group & parent,
+                                                  localIndex const numConstitutivePointsPerParentIndex )
+{
+  DiffusionBase::allocateConstitutiveData( parent, numConstitutivePointsPerParentIndex );
+
+  for( localIndex ei = 0; ei < parent.size(); ++ei )
```

Comment:
I would move this to a function called `initialize() `and write a device kernel for it.

Reply:
Will do!